plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt

```diff
@@ -1268,6 +1260,37 @@ class ClassFileToSourceStubConverter(val kaptContext: KaptContextForStubGenerati
             return treeMaker.Select(typeExpression, treeMaker.name("class"))
         }
 
+        fun unwrapArgumentExpression() =
+            when (singleArg) {
+                is KtCallExpression -> {
+                    val resultingDescriptor = singleArg.getResolvedCall(kaptContext.bindingContext)?.resultingDescriptor
+
+                    if (resultingDescriptor is FunctionDescriptor && resultingDescriptor.fqNameSafe.asString() == "kotlin.arrayOf")
```

Comment:
Do we have a test with `arrayOf()`?

Reply:
Also, what about `intArrayOf()`, `booleanArrayOf()` and other similar functions?

Reply:
Also, you can avoid hard-coding the function fqn. All names are listed in `org.jetbrains.kotlin.resolve.ArrayFqNames`.

Reply:
I just extracted function from existing code ;). But ok, I'll have a look and fix